community/ndp/messaging-v1/src/main/java/org/neo4j/ndp/messaging/v1/PackStreamMessageFormatV1.java

```diff
@@ -0,0 +1,580 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.ndp.messaging.v1;
+
+import java.io.IOException;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.neo4j.graphdb.DynamicLabel;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Path;
+import org.neo4j.graphdb.PropertyContainer;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.helpers.collection.Iterables;
+import org.neo4j.ndp.messaging.v1.infrastructure.ValueNode;
+import org.neo4j.ndp.messaging.v1.infrastructure.ValuePath;
+import org.neo4j.ndp.messaging.v1.infrastructure.ValueRelationship;
+import org.neo4j.ndp.messaging.v1.message.Message;
+import org.neo4j.packstream.PackStream;
+import org.neo4j.packstream.PackType;
+import org.neo4j.ndp.runtime.internal.Neo4jError;
+import org.neo4j.stream.Record;
+
+import static org.neo4j.ndp.runtime.internal.Neo4jError.codeFromString;
+import static org.neo4j.stream.Records.record;
+
+public class PackStreamMessageFormatV1 implements MessageFormat
```

Comment:
I actually like having it in both - the package groups the related stuff for that version, and the version in the name clarifies that this is the v1 message format and allows mixing this class with other versions without fully qualified names.